File: day05/q1.py

def solve(input_data: str) -> str | int | float:
	top, bottom = tuple(input_data.split('\n\n'))
	fresh_ranges = [
		(int(a), (int(b)))
		for a, b in map(lambda ln: tuple(ln.split("-")), top.splitlines())
	]
	return sum([
		1 if any(ing_id >= a and ing_id <= b for a, b in fresh_ranges) else 0
		for ing_id in map(lambda x: int(x), bottom.splitlines())
	])

File: day05/q1.ts
export const solve = (inputData: string) => {
	const [top, bottom] = inputData.split('\n\n')
	const ranges = top!.split('\n').map((ln) => ln.split('-').map((i) => Number(i)))
	return bottom!.split('\n').map((ing) => Number(ing)).reduce((count, ing) => (
		ranges.some(([min, max]) => ing >= min! && ing <= max!) ? count + 1 : count
	), 0)
}
